build/api-docs-generator.js

```diff
@@ -0,0 +1,620 @@
+
+  // Remove custom_edit_url lines
+  content = content.replace(/---\s*custom_edit_url:\s*null\s*---/g, '');
+
+  // Add import statement
+  const directoryName = path.basename(dir);
+  const fileNameWithoutExt = path.basename(fileName, '.mdx');
+  const importStatement = `### Import :\n\n\`\`\`\nbrackets.getModule("${directoryName}/${fileNameWithoutExt}")\n\`\`\`\n\n`;
+
+  // Organize content into sections
+  const sections = {
+    methods: [],
+    properties: [],
+    others: []
+  };
+
+  const blocks = content.split(/(?=```ts)/);
+
+  blocks.forEach(block => {
+    if (block.includes('```ts\nfunction')) {
+      sections.methods.push(processBlock(block, 'function'));
+    } else if (block.includes('```ts\nconst')) {
+      sections.properties.push(processBlock(block, 'const'));
+    } else {
+      sections.others.push(processBlock(block, ''));
+    }
+  });
+
+  // Reconstruct the content
+  let newContent = importStatement;
+  if (sections.methods.length > 0) {
+    newContent += '## METHODS\n\n' + sections.methods.join('\n\n');
+  }
+  if (sections.properties.length > 0) {
+    newContent += '\n\n## PROPERTIES\n\n' + sections.properties.join('\n\n');
+  }
+  if (sections.others.length > 0) {
+    newContent += '\n\n## OTHER MEMBERS\n\n' + sections.others.join('\n\n');
+  }
+
+  // Remove all backticks and clean up 'ts' leftovers
+  newContent = removeBackticksAndCleanup(newContent);
+
+  fs.writeFileSync(filePath, newContent);
+}
+
+
+/**
+ * Processes a block of MDX content.
+ * Creates subheadings for the functions, constructors and other members
+ * 
+ * @param {string} block - Block of MDX content
+ * @param {string} type - Type of block (function, const, or empty string)
+ * @returns {string} Processed block
+ */
+function processBlock(block, type) {
+  const lines = block.split('\n');
+  const declarationLine = lines.findIndex(line => line.startsWith('```ts'));
+
+  if (declarationLine !== -1) {
+    if (type) {
+      const match = lines[declarationLine + 1].match(new RegExp(`${type}\\s+([^(]+)`));
+      if (match) {
+        const name = match[1].trim();
+        lines[declarationLine + 1] = lines[declarationLine + 1].replace(`${type} ${name}`, `### ${name}`);
+      }
+    }
+    // Remove the ```ts line
+    lines.splice(declarationLine, 1);
+  }
+
+  return lines.join('\n');
+}
+
+
+/**
+ * Removes backticks and cleans up 'ts' leftovers in MDX content.
+ * 
+ * @param {string} content - MDX content to clean up
+ * @returns {string} Cleaned up content
+ */
+function removeBackticksAndCleanup(content) {
+  // Remove all backticks
+  content = content.replace(/```/g, '');
+
+  // Remove 'ts' at the beginning of a line or after a newline character
+  content = content.replace(/(\n|^)ts\s*/g, '$1');
+
+  return content;
+}
+
+
+/**
+ * Cleans up temporary files and directories created during the MDX generation process.
+ * This leaves us with just the API directory
+ */
+function cleanupTempFiles() {
```

Comment:
This can be changed to just 
```js
fs.rmSync(tempDir, { recursive: true, force: true });
fs.rmSync(devApiDir, { recursive: true, force: true });
fs.rmSync(sourceDir, { recursive: true, force: true });
```

as if you set force=true, we dont need to check if the file exists.